resources/assets/js/helpers/router.ts

```diff
@@ -36,21 +38,33 @@ export const useUrlHash = (): void => {
   }, [location.hash, hashFound]);
 };
 
-export const useRouter = (
-  routes: Routes<any, React.ReactElement>,
-): React.ReactElement | null => {
+export const UseRouter: FunctionComponent<{
```

Comment:
I don't really like the change of UseRouter to a component. Or rather, if it is going to be changed to a component, it should not be named like a hook and should be moved to to the components folder. I'm curious why you made the change in the first place.

Reply:
I remember running into multiple problems with the routes array and adding the intl hook (which needs a FC, and has to have the IntlProvider component as an ancestor) to properly translate the titles. However, I refactored everything which should resolve everything without having to make UseRouter a FC.

Reply:
Nicely done.